### ISSUE TITLE
ConcurrentModificationException when resolving dependencies

### ISSUE THREAD

**Issue by vRallev** [gradle/gradle#15289]
This might be a duplicate of #15120, but this ticket has been closed and the fix has been reverted. 

We recently upgraded Kotlin from `1.3.72` to `1.4.10` in our `buildSrc` directory and see flakes since then. The full stacktrace is:
<details>
  <summary>Stacktrace</summary>
  
  ```
[2020-11-22 08:34:05] FAILURE: Build failed with an exception.
[2020-11-22 08:34:05] 
[2020-11-22 08:34:05] * What went wrong:
[2020-11-22 08:34:05] Execution failed for task ':features:abc:resolveDependenciesTask'.
[2020-11-22 08:34:05] > java.util.ConcurrentModificationException (no error message)
[2020-11-22 08:34:05] 
[2020-11-22 08:34:05] * Try:
[2020-11-22 08:34:05] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[2020-11-22 08:34:05] 
[2020-11-22 08:34:05] * Exception is:
[2020-11-22 08:34:05] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':features:checkout-v2:item-configuration:fake:resolveDependenciesTask'.
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:200)
[2020-11-22 08:34:05] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:198)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:179)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
[2020-11-22 08:34:05] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[2020-11-22 08:34:05] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[2020-11-22 08:34:05] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2020-11-22 08:34:05] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2020-11-22 08:34:05] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[2020-11-22 08:34:05] 	at java.base/java.lang.Thread.run(Thread.java:834)
[2020-11-22 08:34:05] Caused by: java.util.ConcurrentModificationException
[2020-11-22 08:34:05] 	at org.gradle.api.internal.collections.AbstractIterationOrderRetainingElementSource$RealizedElementCollectionIterator.checkForComodification(AbstractIterationOrderRetainingElementSource.java:285)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.collections.AbstractIterationOrderRetainingElementSource$RealizedElementCollectionIterator.next(AbstractIterationOrderRetainingElementSource.java:252)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:479)
[2020-11-22 08:34:05] 	at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:351)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet$DomainObjectCompositeCollection.collect(CompositeDomainObjectSet.java:182)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet$DomainObjectCompositeCollection.iterator(CompositeDomainObjectSet.java:221)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet.iterator(CompositeDomainObjectSet.java:135)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DelegatingDomainObjectSet.iterator(DelegatingDomainObjectSet.java:140)
[2020-11-22 08:34:05] 	at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:351)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet$DomainObjectCompositeCollection.collect(CompositeDomainObjectSet.java:182)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet$DomainObjectCompositeCollection.iterator(CompositeDomainObjectSet.java:221)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet.iterator(CompositeDomainObjectSet.java:135)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DelegatingDomainObjectSet.iterator(DelegatingDomainObjectSet.java:140)
[2020-11-22 08:34:05] 	at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:351)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet$DomainObjectCompositeCollection.collect(CompositeDomainObjectSet.java:182)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.CompositeDomainObjectSet$DomainObjectCompositeCollection.iterator(CompositeDomainObjectSet.java:221)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DefaultDomainObjectCollection.iterator(DefaultDomainObjectCollection.java:130)
[2020-11-22 08:34:05] 	at java.base/java.util.AbstractCollection.toArray(AbstractCollection.java:140)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DelegatingDomainObjectSet.toArray(DelegatingDomainObjectSet.java:165)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DelegatingDomainObjectSet.toArray(DelegatingDomainObjectSet.java:165)
[2020-11-22 08:34:05] 	at java.base/java.util.ArrayList.addAll(ArrayList.java:701)
[2020-11-22 08:34:05] 	at kotlin.collections.CollectionsKt__MutableCollectionsKt.addAll(MutableCollections.kt:116)
[2020-11-22 08:34:05] 	at org.jetbrains.kotlin.gradle.internal.KotlinDependenciesManagementKt.chooseAndAddStdlibDependency(KotlinDependenciesManagement.kt:363)
[2020-11-22 08:34:05] 	at org.jetbrains.kotlin.gradle.internal.KotlinDependenciesManagementKt.access$chooseAndAddStdlibDependency(KotlinDependenciesManagement.kt:1)
[2020-11-22 08:34:05] 	at org.jetbrains.kotlin.gradle.internal.KotlinDependenciesManagementKt$configureStdlibDefaultDependency$$inlined$with$lambda$1$1.invoke(KotlinDependenciesManagement.kt:79)
[2020-11-22 08:34:05] 	at org.jetbrains.kotlin.gradle.internal.KotlinDependenciesManagementKt$configureStdlibDefaultDependency$$inlined$with$lambda$1$1.invoke(KotlinDependenciesManagement.kt)
[2020-11-22 08:34:05] 	at org.jetbrains.kotlin.gradle.internal.KotlinDependenciesManagementKt$sam$org_gradle_api_Action$0.execute(KotlinDependenciesManagement.kt)
[2020-11-22 08:34:05] 	at org.gradle.internal.ImmutableActionSet$SetWithFewActions.execute(ImmutableActionSet.java:285)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.runDependencyActions(DefaultConfiguration.java:452)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.runDependencyActions(DefaultConfiguration.java:459)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.runDependencyActions(DefaultConfiguration.java:459)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.runDependencyActions(DefaultConfiguration.java:459)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$1.call(DefaultConfiguration.java:609)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$1.call(DefaultConfiguration.java:606)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveGraphIfRequired(DefaultConfiguration.java:606)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.lambda$resolveExclusively$4(DefaultConfiguration.java:586)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.project.DefaultProjectStateRegistry$CalculatedModelValueImpl.update(DefaultProjectStateRegistry.java:361)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveExclusively(DefaultConfiguration.java:583)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveToStateOrLater(DefaultConfiguration.java:570)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.access$1800(DefaultConfiguration.java:139)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationResolvableDependencies$LenientResolutionResult.resolve(DefaultConfiguration.java:1630)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationResolvableDependencies$LenientResolutionResult.getRoot(DefaultConfiguration.java:1643)
[2020-11-22 08:34:05] 	at com.squareup.module.structure.dependencies.ProjectDependenciesKt$resolveDependenciesIfNecessary$1.execute(ProjectDependencies.kt:44)
[2020-11-22 08:34:05] 	at com.squareup.module.structure.dependencies.ProjectDependenciesKt$resolveDependenciesIfNecessary$1.execute(ProjectDependencies.kt)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:44)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.DefaultDomainObjectCollection.configureEach(DefaultDomainObjectCollection.java:187)
[2020-11-22 08:34:05] 	at com.squareup.module.structure.dependencies.ProjectDependenciesKt.resolveDependenciesIfNecessary(ProjectDependencies.kt:36)
[2020-11-22 08:34:05] 	at com.squareup.module.structure.tasks.ResolveDependenciesTask.resolveDependencies(ResolveDependenciesTask.kt:30)
[2020-11-22 08:34:05] 	at jdk.internal.reflect.GeneratedMethodAccessor652.invoke(Unknown Source)
[2020-11-22 08:34:05] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2020-11-22 08:34:05] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2020-11-22 08:34:05] 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:555)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:71)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:71)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:540)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:523)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:108)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:271)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:260)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:34)
[2020-11-22 08:34:05] 	at java.base/java.util.Optional.orElseGet(Optional.java:369)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:34)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:42)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
[2020-11-22 08:34:05] 	at java.base/java.util.Optional.map(Optional.java:265)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
[2020-11-22 08:34:05] 	at java.base/java.util.Optional.orElseGet(Optional.java:369)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
[2020-11-22 08:34:05] 	at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:187)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:179)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
[2020-11-22 08:34:05] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
[2020-11-22 08:34:05] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
[2020-11-22 08:34:05] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
[2020-11-22 08:34:05] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
[2020-11-22 08:34:05] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[2020-11-22 08:34:05] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[2020-11-22 08:34:05] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2020-11-22 08:34:05] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2020-11-22 08:34:05] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[2020-11-22 08:34:05] 	at java.base/java.lang.Thread.run(Thread.java:834)
  ```
</details>

The Gradle task itself is very simple and tiny:
```kotlin
  @TaskAction
  fun resolveDependencies() {
      configurations.configureEach { config ->
        val name = config.name.toLowerCase(US)
        if (config.isCanBeResolved &&
          !name.contains("metadata") &&
          !name.contains("release")
        ) {
          // The following line will resolve all transitive dependencies, otherwise we don't get the
          // correct information.
          config.incoming.resolutionResult.root.dependencies
        }
      }
  }
```
We run this task for more than 3k modules in parallel. I tested Kotlin `1.4.20` and the latest Gradle 6.8 nightly from today and can reproduce the bug. It doesn't happen with Kotlin `1.3.72`, which is a little strange and unclear to me why. 

We have quite a few tasks that analyze dependencies of a project and more complex scenarios also fail. 

Is there anything else that you need or we can provide?

**Comment by vRallev** [comment]
I think that's the bad guy:
```
[2020-11-22 08:34:05] 	at org.jetbrains.kotlin.gradle.internal.KotlinDependenciesManagementKt.chooseAndAddStdlibDependency(KotlinDependenciesManagement.kt:363)
```
Kotlin adds the standard library dependency by default since Kotlin 1.4. This function seems to fail, but it's unclear to me whether it's a bug in Gradle itself, because the Kotlin Gradle Plugin works with configurations, or in the Kotlin Gradle Plugin.

**Comment by jbartok** [comment]
Could you pls. tell us what Gradle version have you been using? Also, could you pls. try it with a more recent Gradle version (7.2+)?

**Comment by Tapchicoma** [comment]
Should be fixed in Kotlin 1.8.0 via https://youtrack.jetbrains.com/issue/KT-41642/TaskDependencyResolveException-Could-not-determine-the-dependencies-when-trying-to-apply-stdlib

**Comment by pschichtel** [comment]
Is this fixed these days?
